api/envoy/extensions/filters/http/cache/v3/cache.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.cache.v3;
+
+import "envoy/api/v2/route/route_components.proto";
+import "envoy/type/matcher/v3alpha/string.proto";
+
+import "google/protobuf/any.proto";
+
+import "udpa/annotations/versioning.proto";
+
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.cache.v3";
+option java_outer_classname = "CacheProto";
+option java_multiple_files = true;
+
+// [#protodoc-title: HTTP Cache Filter]
+// HTTP Cache Filter :ref:`overview <config_http_filters_cache>`.
+// [#extension: envoy.extensions.filters.http.cache]
+
+// [#next-free-field: 6]
+message CacheConfig {
+  // [#not-implemented-hide:]
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  message KeyCreatorParams {
+    // If true, exclude the URL scheme from the cache key. Set to true if your origins always
+    // produce the same response for http and https requests.
+    bool exclude_scheme = 1;
+
+    // If true, exclude the host from the cache key. Set to true if your origins' responses don't
+    // ever depend on host.
+    bool exclude_host = 2;
+
+    // If *query_parameters_included* is nonempty, only query parameters matched
+    // by one or more of its matchers are included in the cache key. Any other
+    // query params will not affect cache lookup.
+    repeated envoy.api.v2.route.QueryParameterMatcher query_parameters_included = 3;
```

Comment:
If you are skipping the v2 API and going straight to v3, you will need to reference v3 types here and elsewhere.

Reply:
reverted, merged, and reworked. seems good now.